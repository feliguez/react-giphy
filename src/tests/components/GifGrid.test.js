import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test <GifGrid />', () => {
  const category = 'goku';
  // const wrapper = shallow(<GifGrid category={category} />);

  test('should match with snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should return some values using useFetchGifs', () => {
    const gifs = [
      {
        id: 1234,
        title: 'Testing title',
        url: 'http://localhost/test.jpg',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
