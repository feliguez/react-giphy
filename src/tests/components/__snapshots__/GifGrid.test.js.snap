// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> should match with snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test <GifGrid /> should return some values using useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={1234}
      key="1234"
      title="Testing title"
      url="http://localhost/test.jpg"
    />
  </div>
</Fragment>
`;
